sdk/monitor/monitor-opentelemetry/src/metrics/quickpulse/filtering.ts

```diff
@@ -0,0 +1,561 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import {
+  DerivedMetricInfo,
+  KnownTelemetryType,
+  FilterInfo,
+  KnownPredicateType,
+  CollectionConfigurationError,
+  FilterConjunctionGroupInfo,
+  KnownAggregationType,
+} from "../../generated";
+import { RequestData, TelemetryData, DependencyData, ExceptionData, TraceData } from "./types";
+import {
+  isRequestData,
+  isDependencyData,
+  isExceptionData,
+  isTraceData,
+  getMsFromFilterTimestampString,
+} from "./utils";
+
+export class TelemetryTypeError extends Error {
```

Comment:
It would be cleaner to create a quickpulseErrors.ts and put the error class definitions there